pkg/scheduler/framework/plugins/dynamicresources/dynamicresources.go

```diff
@@ -149,65 +198,37 @@ func (d *stateData) initializePodSchedulingContexts(ctx context.Context, pod *v1
 						Kind:       "Pod",
 						Name:       pod.Name,
 						UID:        pod.UID,
-						Controller: &controller,
+						Controller: pointer.Bool(true),
```

Comment:
can we use `metav1.NewControllerRef`?

Reply:
That looks neat. There's just one small difference: should `BlockOwnerDeletion` be set? 
`NewControllerRef` sets it, the code previously didn't.

Looking at https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1#OwnerReference the answer is probably, "no, don't set it". It requires delete permission for Pods and can delay pod removal.

I'm using it in a helper function which clears `BlockOwnerDeletion`.

Reply:
The user here is scheduler, which has delete permission for Pods, doesn't it?

Do you think it's useful to keep a PodScheduling object if the Pod is gone?

Reply:
> The user here is scheduler, which has delete permission for Pods, doesn't it?

Indeed [it has](https://github.com/kubernetes/kubernetes/blob/4f303d123659fb3aac2108f8439c7f43dde438b5/plugin/pkg/auth/authorizer/rbac/bootstrappolicy/policy.go#L552).

But what for? Ah, probably preemption.

> Do you think it's useful to keep a PodScheduling object if the Pod is gone?

The PodSchedulingContext object will get removed. The only difference is that it's not required to be removed first.

It boils down to user expectations: if someone deletes a pod with foreground deletion, do they want the PodSchedulingContext object to get removed first? My gut feeling is that the user doesn't care and prefers to have the pod gone as quickly as possible, but I don't have any specific statements about this.

Reply:
I think the normal user expectation is that when they use foreground deletion, then any dependency is deleted.

FWIIW, the default is background deletion.

I don't think we should change what we do from other controllers and keep the blockOwnerDeletion set.

Reply:
Okay, I've changed it.